src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,438 @@
+    }
+  }
+
+  return true;
+}
+#endif
+
+void CubeMap::loadTexture(TextureType type,
+                          const std::string& imageFilePrefix,
+                          const std::string& imageFileExtension) {
+  textureTypeSanityCheck(flags_, type, "CubeMap::loadTexture():");
+
+  // plugin manager used to instantiate importers which in turn are used
+  // to load image data
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+  Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+      manager.loadAndInstantiate("AnyImageImporter");
+  CORRADE_INTERNAL_ASSERT(importer);
+
+  const char* coordStrings[6] = {"+X", "-X", "+Y", "-Y", "+Z", "-Z"};
+  int imageSize = 0;
+
+  // set images
+  Mn::GL::CubeMapTexture* texture = nullptr;
+  switch (type) {
+    case TextureType::Color:
+      texture = textures_[TextureType::Color].get();
+      break;
+
+    case TextureType::Depth:
+      texture = textures_[TextureType::Depth].get();
+      break;
+  }
+  CORRADE_ASSERT(texture, "CubeMap::loadTexture(): Unknown texture type.", );
+
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    // open image file
+
+    std::string filename = Cr::Utility::formatString(
+        "{}.{}.{}.{}", imageFilePrefix, getTextureTypeFilenameString(type),
+        coordStrings[iFace], imageFileExtension);
+
+    importer->openFile(filename);
+    Cr::Containers::Optional<Mn::Trade::ImageData2D> imageData =
+        importer->image2D(0);
+
+    // sanity checks
+    CORRADE_INTERNAL_ASSERT(imageData);
+    Mn::Vector2i size = imageData->size();
+    CORRADE_ASSERT(
+        size.x() == size.y(),
+        " CubeMap::loadTexture(): each texture image must be a square.", );
+    if (iFace == 0) {
+      imageSize = size.x();
+      reset(imageSize);
+    } else {
+      CORRADE_ASSERT(size.x() == imageSize,
+                     " CubeMap::loadTexture(): texture images must have the "
+                     "same size.", );
+    }
+
+    switch (type) {
+      case TextureType::Color:
+        texture->setSubImage(convertFaceIndexToCubeMapCoordinate(iFace), 0, {},
+                             *imageData);
+        break;
+
+      case TextureType::Depth: {
+        // The pixel format for depth texture is R32F. When it is saved as hdr,
+        // the single channel is expanded to three channels by repeating the R
+        // channel 3 times (and it becomes RGB32F).
+        // That means when it is loaded, we need to dedup by taking just the
+        // first component (R component) out of each pixel.
+        Cr::Containers::StridedArrayView2D<const Mn::Color3>
+            imageStridedArrayView = imageData->pixels<Mn::Color3>();
+
+        Cr::Containers::Array<float> depthImage{
+            static_cast<size_t>(size.x() * size.y())};
+        unsigned int idx = 0;
+        for (auto row : imageStridedArrayView) {
+          for (const Mn::Color3& pixel : row) {
+            depthImage[idx++] = pixel.r();
+          }
+        }
```

Comment:
If you want to get fancy, it's possible to do the copy in a potentially more efficient way using [Utility::copy()](https://doc.magnum.graphics/corrade/namespaceCorrade_1_1Utility.html#a6cd0c04124a0d7c4879d4cf6da3556a2). It takes two views of the same type, dimensions and size as input, so you need to turn the `imageStridedArrayView` into a float view as well and make a view on the `depthImage`:

```cpp
/* This casts the Color3 view to a float, which effectively uses only four bytes of each element, 
   which is the red channel */
Cr::Containers::StridedArrayView2D<const float> red = 
    Cr::Containers::arrayCast<const float>(imageStridedArrayView);
/* Turns the `depthImage` into a tightly packed 2D view of the same rows and pixels as 
   `imageStridedArrayView` */
Cr::Containers::StridedArrayView2D<float> output{depthImage, {std::size_t(size.y()), std::size_t(size.x())}};
/* And this does a copy of the data */
Cr::Utility::copy(red, output);
```

Be sure to test this, I might have made some minor mistake and caused copy() to assert ;)